### PR TITLE
[ImageCaptureCore] Add missing binding

### PR DESCRIPTION
**Pull request by haritha-mohan**
Deprecated "cameraDevice:didReceiveThumbnailForItem:" for macOS and macCatalyst platforms and added updated binding for "cameraDevice:didReceiveThumbnail:forItem:error:" method. 

Refs:
[Deprecated method](https://developer.apple.com/documentation/imagecapturecore/iccameradevicedelegate/1507941-cameradevice?changes=la_5&language=objc)
[Updated method](https://developer.apple.com/documentation/imagecapturecore/iccameradevicedelegate/3142895-cameradevice?changes=la_5&language=objc)

### CODE REVIEW

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
No xtro todo changes?

File: src/imagecapturecore.cs [hunk 1]
```diff
@@ -456,9 +456,15 @@ interface ICCameraDeviceDelegate : ICDeviceDelegate {
...
 
 		[Abstract]
+		[Deprecated (PlatformName.MacOSX, 10, 15, message: "Use 'DidReceiveThumbnailForItem (ICCameraDevice, CGImageRef, ICCameraItem, NSError)' instead.")]
+		[Deprecated (PlatformName.MacCatalyst, 13, 1, message: "Use 'DidReceiveThumbnailForItem (ICCameraDevice, CGImageRef, ICCameraItem, NSError)' instead.")]
 		[Export ("cameraDevice:didReceiveThumbnailForItem:")]
 		void DidReceiveThumbnail (ICCameraDevice camera, ICCameraItem forItem);
 
+		[Abstract]
+		[Export ("cameraDevice:didReceiveThumbnail:forItem:error:")]
+		void DidReceiveThumbnailForItem (ICCameraDevice camera, /* CGImageRef */ IntPtr thumbnail, ICCameraItem forItem, [NullAllowed] NSError error);
```
Comment (dalexsoto) [inline src/imagecapturecore.cs:466]: Can't we do CGImage here? 

```suggestion
		void DidReceiveThumbnailForItem (ICCameraDevice camera, CGImage thumbnail, ICCameraItem forItem, [NullAllowed] NSError error);
```

**Review by dalexsoto — CHANGES_REQUESTED** [review]

File: src/imagecapturecore.cs (hunk 1 above)
Comment (haritha-mohan) [reply src/imagecapturecore.cs:466]: @dalexsoto I thought about that, but after reading [this](https://stackoverflow.com/questions/30909126/why-use-cgimageref-not-cgimage-instead) and looking at how we handled CGImageRef in other parts of the codebase, I thought this implementation was better. But if the overall consensus is that using CGImage is a better option, I can change it.

File: src/imagecapturecore.cs [hunk 2]
```diff
@@ -456,9 +456,15 @@ interface ICCameraDeviceDelegate : ICDeviceDelegate {
 		void DidChangeCapability (ICCameraDevice camera);
 
 		[Abstract]
+		[Deprecated (PlatformName.MacOSX, 10, 15, message: "Use 'DidReceiveThumbnailForItem (ICCameraDevice, CGImageRef, ICCameraItem, NSError)' instead.")]
+		[Deprecated (PlatformName.MacCatalyst, 13, 1, message: "Use 'DidReceiveThumbnailForItem (ICCameraDevice, CGImageRef, ICCameraItem, NSError)' instead.")]
 		[Export ("cameraDevice:didReceiveThumbnailForItem:")]
 		void DidReceiveThumbnail (ICCameraDevice camera, ICCameraItem forItem);
 
+		[Abstract]
```
Comment (rolfbjarne) [inline src/imagecapturecore.cs:464]:
```suggestion
#if XAMCORE_5_0
		[Abstract]
#endif
```

Comment (rolfbjarne) [inline src/imagecapturecore.cs:464]:
```suggestion
#if XAMCORE_5_0
		[Abstract]
#endif
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Comment by rolfbjarne** [comment]
@mandel-macaque 

> No xtro todo changes?

No, but with this, yes: https://github.com/xamarin/xamarin-macios/pull/17417

**Comment by rolfbjarne** [comment]
@haritha-mohan 

> @mandel-macaque
> 
> > No xtro todo changes?
> 
> No, but with this, yes: #17417

So I just merged that PR, which means that if you merge main into your branch, you'll now have an xtro failure you need to fix as well.

**Review by rolfbjarne — APPROVED** [review]

File: src/imagecapturecore.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/imagecapturecore.cs:471]: @dalexsoto we are dealing with a delegate in this case.

**Review by mandel-macaque — APPROVED** [review]

File: src/imagecapturecore.cs (hunk 1 above)
Comment (dalexsoto) [reply src/imagecapturecore.cs:471]: Yeah also `CGImageRef` and `CGImage` seem not to be toll-free bridged so this is ok!

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**